flink-ml-lib/src/test/java/org/apache/flink/ml/classification/knn/KnnTest.java

```diff
@@ -0,0 +1,287 @@
+        KnnModel model = knn.fit(testData);
+        Table output = model.transform(testData)[0];
+
+        assertEquals(
+                Arrays.asList("label", "features", "prediction"),
+                output.getResolvedSchema().getColumnNames());
+        executeAndCollect(output);
+    }
+
+    @Test
+    public void testFitAndPredict() throws Exception {
+        Knn knn =
+                new Knn()
+                        .setLabelCol("label")
+                        .setFeaturesCol("features")
+                        .setK(4)
+                        .setPredictionCol("prediction");
+        KnnModel knnModel = knn.fit(trainData);
+        Table output = knnModel.transform(testData)[0];
+        List<Tuple2<Double, Double>> result = executeAndCollect(output);
+        verifyClusteringResult(result);
+    }
+
+    @Test
+    public void testSaveLoadAndPredict() throws Exception {
+        String path = Files.createTempDirectory("").toString();
+        Knn knn =
+                new Knn()
+                        .setLabelCol("label")
+                        .setFeaturesCol("features")
+                        .setK(4)
+                        .setPredictionCol("prediction");
+        knn.save(path);
+
+        Knn loadKnn = Knn.load(env, path);
+        KnnModel knnModel = loadKnn.fit(trainData);
+        Table output = knnModel.transform(testData)[0];
+
+        List<Tuple2<Double, Double>> result = executeAndCollect(output);
+        verifyClusteringResult(result);
+    }
+
+    @Test
+    public void testModelSaveLoadAndPredict() throws Exception {
+        String path = Files.createTempDirectory("").toString();
+        Knn knn =
+                new Knn()
+                        .setLabelCol("label")
+                        .setFeaturesCol("features")
+                        .setK(4)
+                        .setPredictionCol("prediction");
+        KnnModel knnModel = knn.fit(trainData);
+        knnModel.save(path);
+        env.execute();
+
+        KnnModel newModel = KnnModel.load(env, path);
+        Table output = newModel.transform(testData)[0];
+        List<Tuple2<Double, Double>> result = executeAndCollect(output);
+        verifyClusteringResult(result);
+    }
+
+    @Test
+    public void testGetModelData() throws Exception {
+        Knn knn =
+                new Knn()
+                        .setLabelCol("label")
+                        .setFeaturesCol("features")
+                        .setK(4)
+                        .setPredictionCol("prediction");
+
+        KnnModel knnModel = knn.fit(trainData);
+        Table modelData = knnModel.getModelData()[0];
+
+        DataStream<Row> output = tEnv.toDataStream(modelData);
+        assertEquals("f0", modelData.getResolvedSchema().getColumnNames().get(0));
+
+        List<Row> modelRows = IteratorUtils.toList(output.executeAndCollect());
+
+        KnnModelData data = (KnnModelData) modelRows.get(0).getField(0);
+
+        assert data != null;
```

Comment:
nits: Could we use `Assert.assertNotNull(...)` for consistency. with `Assert.assertEquals(...)` used below?

Reply:
OK